client/src/pages/LoginOrSignUp.js

```diff
@@ -0,0 +1,136 @@
+import React, { useState, useContext } from "react";
+import { Link } from "react-router-dom";
+import { Container, Paper, Typography, makeStyles } from "@material-ui/core";
+import { AuthContext } from "../context/AuthContext";
+import FormInput from "../components/FormInput";
+import GenericButton from "../components/GenericButton";
+
+function LoginOrSignUp({ type }) {
+  const classes = useStyles();
+  const { errors, registerUser, loginUser } = useContext(AuthContext);
+  const [credentials, setCredentials] = useState({
+    name: "",
+    email: "",
+    password: "",
+  });
+
+  function handleChange(event) {
+    const { id, value } = event.target;
+    setCredentials((prev) => ({ ...prev, [id]: value }));
+  }
+
+  function handleSubmit(event) {
+    event.preventDefault();
+    if (type === "register") {
```

Comment:
Maybe not a good idea to handle login and signup in a single component. It's better to create separate Login.js and Signup.js

Reply:
I combined login and signup into a single component to try avoiding writing redundant code (since the UI is basically the same, except for one having an extra input to handle).